tests/unit/test_grid.py

```diff
@@ -0,0 +1,287 @@
+"""Tests for grid module."""
+
+from pathlib import Path
+
+import numpy as np
+import pytest
+import xarray as xr
+from xarray import DataArray, DataTree
+
+from smap_l2_gridder.grid import (
+    default_fill_value,
+    get_grid_information,
+    get_target_grid_information,
+    grid_variable,
+    prepare_variable,
+    process_input,
+    transfer_metadata,
+    variable_fill_value,
+)
+
+
+# Fixtures
+@pytest.fixture
+def sample_datatree(tmp_path):
+    """Create a sample DataTree for testing.
+
+    It is round tripped to a temporary disk location for easy of setting the
+    correct NetCDF attributes.
+
+    This represents the expected shape of an L2_SMPE granule.
```

Comment:
Is L2_SMPE an intentional abbreviation of all the SMAP collections?

Reply:
Nope, it's the shortname inside one of the input granules `SMAP_L2_SM_P_E_03107_D_20150901T002911_R19240_001.h5`


Update it's what I thought it was in there 
it's actually `string :shortName = "SPL2SMP_E" ;`
but also 
`string :SMAPShortName = "L2_SM_P_E" ;`

Reply:
1b5e06c